akka-actor-typed/src/main/scala/akka/actor/typed/Behavior.scala

```diff
@@ -261,16 +260,26 @@ object Behavior {
    * INTERNAL API: When the cell is stopping this behavior is used, so
    * that PostStop can be sent to previous behavior from `finishTerminate`.
    */
-  private[akka] class StoppedBehavior[T](val postStop: OptionVal[Behavior[T]]) extends Behavior[T] {
-    validatePostStop(postStop)
+  private[akka] sealed class StoppedBehavior[T](val postStop: OptionVal[TypedActorContext[T] ⇒ Unit]) extends ExtensibleBehavior[T] {
```

Comment:
This is half of the essence of this PR - make StoppedBehavior an extensible behavior, and implement the callback as a normal signal (which can come through the interceptor)